src/main.rs

```diff
@@ -120,11 +40,20 @@ fn main() -> Result<()> {
             }
         };
 
-        let file_name = path.file_name().unwrap();
+        let path = Path::new(&args.outdir);
+        if path.is_file() {
+            return Err(anyhow::anyhow!(
+                "Can't create directory: a file with the same name already exists!"
+            ));
+        } else if path.is_dir() {
+        } else {
+            std::fs::create_dir(&args.outdir)?;
+        }
```

Comment:
```suggestion
        if path
            .try_exists()
            .context("Couldn't determine if file already exists")?
        {
            ensure!(
                path.is_dir(),
                "Can't create directory: a file with the same name already exists!"
            );
        } else {
            std::fs::create_dir(&args.outdir)?;
        }
```

Not entirely sure if this handles symlinks and other types properly. A better solution may be to check for existence instead.

Anyhow's `ensure!(x, "foo")` can be used to simplify the `if !x { return anyhow!("foo"); }` pattern.
`context()` can be used to add additional context to errors before bubbling them up (for better error messages).

Reply:
```rust
    match std::fs::metadata(path) {
        Ok(m) if m.is_dir() => {}
        Err(e) if e.kind() == std::io::ErrorKind::NotFound => std::fs::create_dir(&args.outdir)?,

        Ok(_) => bail!("Can't create directory: a file with the same name already exists!"),
        Err(e) => Err(e).context("Failed to check out_dir")?,
    }
```
try this I guess lol